accounting/accounting_cli_functions.py

```diff
@@ -0,0 +1,108 @@
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+import sqlite3
+import argparse
+import time
+import sys
+
+import pandas as pd
+
+def view_user(user):
+    try:
+        # open connection to database
+        conn = sqlite3.connect("FluxAccounting.db")
+        # get the information pertaining to a user in the Accounting DB
+        select_stmt = (
+            "SELECT * FROM association_table where user_name='" + user + "'"
+        )
+        dataframe = pd.read_sql_query(select_stmt, conn)
+        # if the length of dataframe is 0, that means
+        # the user specified was not found in the table
+        if len(dataframe.index) == 0:
+            print("User not found in association_table")
+        else:
+            print(dataframe)
+    except pd.io.sql.DatabaseError as e_database_error:
+        print(e_database_error)
+
+def add_user(user_acct_info):
+    # open connection to database
+    conn = sqlite3.connect("FluxAccounting.db")
+
+    # insert the user values into the database
+    try:
+        conn.execute(
+            """
+            INSERT INTO association_table (
+                creation_time,
+                mod_time,
+                deleted,
+                user_name,
+                admin_level,
+                account,
+                parent_acct,
+                shares,
+                max_jobs,
+                max_wall_pj
+            )
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+            """,
+            (
+                int(time.time()),
+                int(time.time()),
+                0,
+                user_acct_info[0],
+                user_acct_info[1],
+                user_acct_info[2],
+                user_acct_info[3],
+                user_acct_info[4],
+                user_acct_info[5],
+                user_acct_info[6],
+            ),
+        )
+        # commit changes
+        conn.commit()
+    # make sure entry is unique
+    except sqlite3.IntegrityError as integrity_error:
+        print(integrity_error)
+    # make sure all fields are entered
+    except IndexError:
+        print(
+            "missing fields for account."
+            "\n\nFIELDS=user_name admin_level account parent_acct shares max_jobs max_wall_pj"
+        )
+
+def delete_user(user):
+    # open connection to database
+    conn = sqlite3.connect("FluxAccounting.db")
+    # delete user account from association_table
+    delete_stmt = (
+        "DELETE FROM association_table WHERE user_name='" + user + "'"
+    )
+    conn.execute(delete_stmt)
+    # commit changes
+    conn.commit()
+
+def edit_user(user_info):
+    # open connection to database
+    conn = sqlite3.connect("FluxAccounting.db")
+    try:
+        # edit value in accounting database
+        conn.execute(
+            "UPDATE association_table SET %s=%s WHERE user_name='%s'"
+            % (user_info[1], user_info[2], user_info[0])
+        )
```

Comment:
Same comment about parameter substitution:
```suggestion
        conn.execute(
            "UPDATE association_table SET ?=? WHERE user_name='?'",
            (user_info[1], user_info[2], user_info[0])
        )
```

Reply:
Note, that my suggestion for this specific query may not work due to the first `?`.  It seems parameterization doesn't work for everything in a SQL statement, like table names [[1]](https://stackoverflow.com/a/597198).  Per that same SO thread:

> For situations where user input determines the table, I pull the name of the table out of a dict and raise an exception if the contents of the input are unexpected. Probably not the best way to go about it, but seems less liable to result in SQL injection.